File: backend/src/HousingApp.Application/Room/DTOs/UpdateRoomDto.cs
using HousingApp.Application.Room.Upload;

namespace HousingApp.Application.Room.DTOs;

public record UpdateRoomDto(
    int RoomId,
    string Name,
    string Description,
    double Latitude,
    double Longitude,
    double Price,
    int RoomStatusId,
    List<ImageRoomUpload> NewImages,
    List<int> KeptImageIds,
    List<CreateRoomPolicyDto> Policies,
    List<CreateRoomServiceDto> Services
);

File: frontend/src/features/edit-room/types/updateRoomDto.ts
export interface UpdateRoomDto {
	roomId: string;
	name: string;
	latitude: number;
	longitude: number;
	description: string;
	price: number;
	roomStatus: number;
	imageRoomFiles: File[];
	keptImageIds: number[];
	services: number[];
	policies: { id: number; description: string }[];
}
